Updating operationId to group the operations in a logical way  …

rahuls-microsoft opened the pull request:
…and generate APIs, so that they are grouped together based on the operations.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

rahuls-microsoft committed:
Updating operationId to group the opearions in a logical way and the generate APIs are grouped together based on the operations.

AutorestCI commented:
/azp run automation - sdk

azuresdkci commented:
Can one of the admins verify this patch?

rahuls-microsoft committed:
1. Updating schema for autoQuotaRequest
2. Updating examples.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Updates to data model and examples.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Model Update.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Data Model Fix.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft commented:
For Error: R3007 - PutGetPatchResponseSchema
/subscriptions/{subscriptionId}/providers/Microsoft.Capacity/resourceProviders/{providerId}/locations/{location}/serviceLimits/{resourceName} has different responses for PUT/GET/PATCH operations. The PUT/GET/PATCH operations must have same schema response.
  at specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:37

The response for get is Quota information.
The response for Put/Patch Quota request contains the result:  provisioningState and then the updated quota information. There is no change is this section.
"properties": {
          "provisioningState": "Succeeded",
          "message": "",
          "properties": {
            "properties": {
              "limit": 200,
              "unit": "Count",
              "currentValue": 160,
              "name": {
                "value": "standardFSv2Family",
                "localizedValue": "Standard FSv2 Family vCPUs"
              },
              "quotaPeriod": "",
              "properties": {}
            }
          }

rahuls-microsoft committed:
Fixing Data Model for AQI.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Removing duplicate definitions.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Adding type=object for emailAction.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Putting the similar definitions together.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Model Fix.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Fixing AQI examples.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Adding descriptions for enum.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Fixing Spell Check error.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft commented:
Hi Reviewers,
Minor changes to group the operations in a logical fashion. The generated Doc based on OperationId wasn't user-friendly. We need to regenerate the documentation for External customers - Adobe, etc. Please prioritize the review.  I have also added the Previous PR reference: #8480
Please let me know, if there is anything needed from my side.

Thanks.

sanjaiganesh commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -87,11 +87,11 @@
         "summary": "Submits a Quota Request for a resource provider at the specified location for the specific resource in the parameter.",
         "description": "Submits Quota change request for a resource provider for the specified location for the specific resource in the parameter. To use, first make a Get request to get quota information for the specific resource. This information consists of information regarding that specific resources. For the specific resource, if it requires an update to the quota, update the limit field i ...
-        "operationId": "QuotaRequest_Create",
+        "operationId": "Quota_Create",
```
>Quota_Create [](start = 24, length = 12)

_CreateOrUpdate

rahuls-microsoft replied:
The Patch is for Update. Quota_Update.
Put - Quota_Create

Does it sound fine?

Breaking Changes:
There are no active users.
We are currently in Private preview, so only internal users. I will coordinate with them, after update.

There are no users for autoQuotaIncrease, so making breaking changes in data model.

Thanks.

sanjaiganesh replied:
Customers can update using PUT as well, if there are writable properties.  please see other specs ..this is the convention used.

---
In reply to: [413977335](https://github.com/Azure/azure-rest-api-specs/pull/9159#discussion_r413977335) [](ancestors = 413977335)

rahuls-microsoft replied:
The Patch is for Update. Quota_Update.
Put - Quota_Create

There is no active user.
We are currently in Private preview, so only internal users. I will coordinate with them, after update.

There are no users for autoQuotaIncrease, so making breaking changes in data model.

Thanks.

rahuls-microsoft replied:
That's right.
Should I apply same opeartionId - Quota_CreateOrUpdate to both Put and patch?
Please suggest.

sanjaiganesh commented:
>     },

this is not used any more

---
Refers to: specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:1072 in e5041dc. [](commit_id = e5041dcb749e41e1ecc6373596a5deebabb856ec, deletion_comment = False)

sanjaiganesh commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -1043,49 +1026,46 @@
       "type": "object",
       "properties": {
         "emailActions": {
           "description": "The email actions for auto quota increase.",
```
these are breaking changes to existing versions. how are we handling conversions between existing resources users created ?

rahuls-microsoft replied:
Breaking Changes:
There are no active users.
We are currently in Private preview, so only internal users. I will coordinate with them, after update.

There are no users for autoQuotaIncrease, so making breaking changes in data model.

Thanks.

deepanjanbhol replied:
There are no users for autoQuotaIncrease api, so making breaking changes in data model.

rahuls-microsoft commented:
> > ```
> > },
> > ```
> 
> this is not used any more
> 
> Refers to: specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:1072 in [e5041dc](https://github.com/Azure/azure-rest-api-specs/commit/e5041dcb749e41e1ecc6373596a5deebabb856ec). [](commit_id = [e5041dc](https://github.com/Azure/azure-rest-api-specs/commit/e5041dcb749e41e1ecc6373596a5deebabb856ec), deletion_comment = False)

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Removing used object: phoneActions.
Updating the comments for Put and Patch, that both can be used to update quota.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft commented:
> > ```
> > },
> > ```
> 
> this is not used any more
> 
> Refers to: specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:1072 in [e5041dc](https://github.com/Azure/azure-rest-api-specs/commit/e5041dcb749e41e1ecc6373596a5deebabb856ec). [](commit_id = [e5041dc](https://github.com/Azure/azure-rest-api-specs/commit/e5041dcb749e41e1ecc6373596a5deebabb856ec), deletion_comment = False)

Removed it.

rahuls-microsoft committed:
Updating the Put operatonId as _CreateOrUpdate.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft commented:
> Previous PR reference: #8480
> 
> For Error: R3007 - PutGetPatchResponseSchema
> /subscriptions/{subscriptionId}/providers/Microsoft.Capacity/resourceProviders/{providerId}/locations/{location}/serviceLimits/{resourceName} has different responses for PUT/GET/PATCH operations. The PUT/GET/PATCH operations must have same schema response.
> at specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:37
> 
> The response for get is Quota information.
> The response for Put/Patch Quota request contains the result: provisioningState and then the updated quota information. There is no change is this section.
> "properties": {
> "provisioningState": "Succeeded",
> "message": "",
> "properties": {
> "properties": {
> "limit": 200,
> "unit": "Count",
> "currentValue": 160,
> "name": {
> "value": "standardFSv2Family",
> "localizedValue": "Standard FSv2 Family vCPUs"
> },
> "quotaPeriod": "",
> "properties": {}
> }
> }

Hi ARM Team,
How do we get pass this error and the breaking change error?
Would it possible to merge it, as it's approved by ARM team (ARMSignedOff ).

Thanks.

rahuls-microsoft commented:
@weidongxu-microsoft Please take a look and suggest how we can merge the changes.

I have explained the reason for both failures:
**Breaking Change:**
Currently on Active customers using the autoQuotaIncrease API, so no Issues.
For quota APIs, only few internal teams, we will coordinate the changes with them. The changes are based on their feedback. So, it's expected.

**Lint Error**
Previous PR reference: #8480
**No Changes from Previous PR.**
For Error: R3007 - PutGetPatchResponseSchema
/subscriptions/{subscriptionId}/providers/Microsoft.Capacity/resourceProviders/{providerId}/locations/{location}/serviceLimits/{resourceName} has different responses for PUT/GET/PATCH operations. The PUT/GET/PATCH operations must have same schema response.
at specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:37

The response for get is Quota information.
The response for Put/Patch Quota request contains the result: provisioningState and then the updated quota information. There is no change is this section.
"properties": {
"provisioningState": "Succeeded",
"message": "",
"properties": {
"properties": {
"limit": 200,
"unit": "Count",
"currentValue": 160,
"name": {
"value": "standardFSv2Family",
"localizedValue": "Standard FSv2 Family vCPUs"
},
"quotaPeriod": "",
"properties": {}
}
}
Please let's expedite the merge, as we need the SDK to share with the teams by 5/1.
Thanks for your help.

weidongxu-microsoft commented on specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-07-19/quota.json:
```diff
@@ -87,11 +87,11 @@
       "put": {
         "tags": [
-          "QuotaRequests"
+          "Quota_CreateOrUpdate"
```
Any reason why tag for same resource being different for different http method (QuotaInformation for get, QuotaRequest_Create for put)? Usually they are same.

rahuls-microsoft replied:
I will make thems ame.

rahuls-microsoft replied:
Updated to QuotaInformation .

weidongxu-microsoft reviewed (commented):
Usually `provisioningState` is also included in `get`.

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/examples/GetVirtualMachine.json#L94

weidongxu-microsoft commented:
Reference of Lint error https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#r3007

rahuls-microsoft committed:
Making Tag information same - QuotaInformation.

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft commented:
@weidongxu-microsoft 
"Usually provisioningState is also included in get.

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/examples/GetVirtualMachine.json#L94"

In this case - GetVirtualMachine - The VM was getting created, if I am correct. That's why the  provisioningState makes sense.

In case of GetQuota - Just gets the Quota information rom Compute. It is a Read operation.
We can put provisioningState =Succeeded. But it was a readonly operation. Would it make sense?

weidongxu-microsoft commented:
/azp run

weidongxu-microsoft commented:
@akning-ms 

Author confirmed that BreakingChange is intended, and Lint problem is reviewed through ARM.

Please help merge.

weidongxu-microsoft commented:
/azp run

weidongxu-microsoft commented:
/azp run

weidongxu-microsoft commented:
@akning-ms 

There is some problem with CI as well. I restarted it multiple time, but it is not running.

akning-ms commented:
/azp run

akning-ms commented:
@NullMDR Can you help take a look why pipeline is always pending?

NullMDR commented:
/azp run

NullMDR commented:
/azp run

AutorestCI commented:
/azp run automation - sdk

rahuls-microsoft committed:
Updating example for commit.

AutorestCI commented:
/azp run automation - sdk

NullMDR commented:
/azp run automation - sdk
